clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go

```diff
@@ -304,16 +304,41 @@ func (w *waitForControlledPodsRunningMeasurement) handleObject(clients *framewor
 	if err := w.deleteObjectLocked(clients.GetClient(), oldRuntimeObj); err != nil {
 		glog.Errorf("%s: delete checker error: %v", w, err)
 	} else if newRuntimeObj == nil {
-		if err := w.handleObjectLocked(clients.GetClient(), oldRuntimeObj, true); err != nil {
+		if err := w.handleObjectLocked(clients.GetClient(), oldRuntimeObj, true, true); err != nil {
```

Comment:
I'm not a fan of passing a list of bools, as this is hard to follow.

I think it would be clear if handleObjectLocked would take both old and new version and encapsulate that logic in the handleObjectLock function?

Reply:
Done.

Reply:
As you suggested, I moved most of the logic to the handleObjectLocked function.